test/e2e/flowaggregator_test.go

```diff
@@ -1581,6 +1598,64 @@ func createToExternalTestServer(t *testing.T, data *TestData) *PodIPs {
 	return serverIPs
 }
 
+func testL7VisibilityControllerRun(t *testing.T, data *TestData) {
+	skipIfFeatureDisabled(t, features.L7Visibility, true, false)
+
+	clientPodName := "test-l7-visibility"
+	clientPodLabels := map[string]string{"test-l7-visibility-e2e": "true"}
+	clientPodAnnotations := map[string]string{antreaagenttypes.L7VisibilityAnnotationKey: "both"}
+	cmd := []string{"sleep", "3600"}
+
+	// Create a client Pod which will be selected by test L7 NetworkPolices.
+	require.NoError(t, NewPodBuilder(clientPodName, data.testNamespace, toolboxImage).OnNode(nodeName(0)).WithCommand(cmd).WithLabels(clientPodLabels).WithAnnotations(clientPodAnnotations).Create(data))
+	clientPodIP, err := data.podWaitForIPs(defaultTimeout, clientPodName, data.testNamespace)
+	if err != nil {
+		t.Fatalf("Error when waiting for IP for Pod '%s': %v", clientPodName, err)
+	}
```

Comment:
better to use `require.NoError` here as well. It is less verbose and consistent with the rest of the test

Reply:
I differ to agree, in this case we need the clientPodIP, I know we can get it later but that is redundant work when we already have a function that checks for Pod IP and returns it.

Reply:
changed with require.Errorf